modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.ignite.internal.processors.task.GridInternal;
+import org.apache.ignite.internal.processors.task.GridVisorManagementTask;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.internal.visor.VisorJob;
+import org.apache.ignite.internal.visor.VisorOneNodeTask;
+import org.apache.ignite.lang.IgniteUuid;
+import org.apache.ignite.spi.systemview.view.SystemView;
+import org.apache.ignite.spi.systemview.view.SystemViewRowAttributeWalker;
+import org.apache.ignite.spi.systemview.view.SystemViewRowAttributeWalker.AttributeWithValueVisitor;
+import org.jetbrains.annotations.Nullable;
+
+import static org.apache.ignite.internal.processors.metric.impl.MetricUtils.toSqlName;
+import static org.apache.ignite.internal.visor.systemview.VisorSystemViewTask.SimpleAttributeType.DATE;
+import static org.apache.ignite.internal.visor.systemview.VisorSystemViewTask.SimpleAttributeType.NUMBER;
+import static org.apache.ignite.internal.visor.systemview.VisorSystemViewTask.SimpleAttributeType.STRING;
+
+/**
+ * Reperesents visor task for obtaining system view content.
+ */
+@GridInternal
+@GridVisorManagementTask
+public class VisorSystemViewTask extends VisorOneNodeTask<VisorSystemViewTaskArg, VisorSystemViewTaskResult> {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** {@inheritDoc} */
+    @Override protected VisorJob<VisorSystemViewTaskArg, VisorSystemViewTaskResult> job(VisorSystemViewTaskArg arg) {
+        return new VisorSystemViewJob(arg, false);
+    }
+
+    /**
+     *
+     */
+    private static class VisorSystemViewJob extends VisorJob<VisorSystemViewTaskArg, VisorSystemViewTaskResult> {
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /**
+         * Create job with specified argument.
+         *
+         * @param arg Job argument.
+         * @param debug Flag indicating whether debug information should be printed into node log.
+         */
+        protected VisorSystemViewJob(@Nullable VisorSystemViewTaskArg arg, boolean debug) {
+            super(arg, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected VisorSystemViewTaskResult run(@Nullable VisorSystemViewTaskArg arg) throws IgniteException {
+            GridSystemViewManager sysViewMgr = ignite.context().systemView();
+
+            String targetSysViewName = arg.systemViewName();
+
+            SystemView<?> targetSysView = sysViewMgr.view(targetSysViewName);
+
+            if (targetSysView == null) { // In this case we assume that the requested system view name is in SQL format.
+                for (SystemView<?> sysView : sysViewMgr) {
+                    if (toSqlName(sysView.name()).equals(targetSysViewName)) {
+                        targetSysView = sysView;
+
+                        break;
+                    }
+                }
+            }
+
+            if (targetSysView == null)
+                return null;
```

Comment:
I advise you to separate the logic for finding a view into a separate method.
In this case, code here will look something like 
```
SystemView<?> targetSysView = findSysView(arg.systemViewName());

            if (targetSysView == null)
                return null;
```

`findSysView` may look something like

```
private SystemView<?> findSysView(String name) {
            GridSystemViewManager sysViewMgr = ignite.context().systemView();

            SystemView<?> targetSysView = sysViewMgr.view(name);

            if (targetSysView != null)
                return targetSysView;

            // comment about sql...
            for (SystemView<?> sysView : sysViewMgr) {
                if (toSqlName(sysView.name()).equals(name))
                    return sysView;
            }

            return null;
        }
```

Reply:
Done.